Source/DafnyDriver/Commands/DocCommand.cs

```diff
@@ -0,0 +1,50 @@
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Linq;
+
+namespace Microsoft.Dafny;
+
+class DocCommand : ICommandSpec {
+
+  public static readonly Option<string> DocProgramNameOption = new("--doc-program-name",
+    "The text to use as program name in generated documentation"
+  );
+
+  public static readonly Option<string> DocFilenameFormat = new("--doc-file-name",
+    "Form of file references in documentation: none, absolute, name (the default), relative=<prefix>"
+  );
+
+  public static readonly Option<bool> DocShowModifyTime = new("--doc-modify-time",
+    "If enabled, includes the lasst modified time of source files in the output"
+  );
+
+
+  public static IEnumerable<Option> DocOptions => new Option[] {
+    CommonOptionBag.Verbose,
+    CommonOptionBag.Output,
+    DocProgramNameOption,
+    DocFilenameFormat
+  }.Concat(ICommandSpec.ResolverOptions);
+
+  static DocCommand() {
+    DafnyOptions.RegisterLegacyBinding(DocProgramNameOption, (options, value) => { options.DocProgramNameOption = value; });
```

Comment:
Instead of these `RegisterLegacyBinding` calls and related fields that were added to DafnyOptions, you can use `DafnyOptions.Get(DocProgramNameOption)` to access the value of these options. `RegisterLegacyBinding` is meant to be used to work with existing (legacy) code.

Reply:
Thanks - I was wondering if there was a way to avoid that connection.